src/util/system.cpp

```diff
@@ -64,6 +64,16 @@
 #include <thread>
 #include <typeinfo>
 
+#ifndef WIN32
+namespace {
+constexpr mode_t private_umask = 0077;
```

Comment:
Is there a reason to add 10 lines here, and another `#ifndef`, instead of putting the 2 lines of code where they are actually used? i.e why not just:
```cpp
#ifndef WIN32
    constexpr mode_t private_umask = 0077;
    umask(private_umask);
#endif
```
in `SetupEnvironment()`?

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/17127#issuecomment-1417887067).